server/cli/launch.py

```diff
@@ -183,4 +187,7 @@ def launch(
         f = open(devnull, "w")
         sys.stdout = f
 
-    server.app.run(host=host, debug=debug, port=port, threaded=True)
+    try:
+        server.app.run(host=host, debug=debug, port=port, threaded=True)
+    except OSError as e:
```

Comment:
there are other reasons you may get an OSError throw. This catch makes them all seem like port selection errors.

Reply:
I added an error number check to fix this.